source/compute/mus_compute_d3q19_module.fpp

```diff
@@ -4346,6 +4355,1120 @@ end subroutine f_f_eq_regularized_4th_ord_d3q19
+        !> input  pdf vector
+        real(kind=rk), intent(in)  ::  inState(nElems * varSys%nScalars)
+        !> output pdf vector
+        real(kind=rk), intent(out) :: outState(nElems * varSys%nScalars)
+        !> Auxiliary field computed from pre-collision state
+        !! Is updated with correct velocity field for multicomponent models
+        real(kind=rk), intent(inout) :: auxField(nElems * varSys%nAuxScalars)
+        !> connectivity vector
+        integer, intent(in) :: neigh(nElems * layout%fStencil%QQ)
+        !> number of elements solved in kernel
+        integer, intent(in) :: nSolve
+        !> current level
+        integer,intent(in) :: level
+        !> global parameters
+        type(mus_param_type),intent(in) :: params
+        !> position of derived quantities in varsys for all fields
+        type( mus_derVarPos_type ), intent(in) :: derVarPos(:)
+        ! -------------------------------------------------------------------- !
+        integer :: iElem, iDir, invDir
+        type(mus_varSys_data_type), pointer :: fPtr
+        type(mus_scheme_type), pointer :: scheme
+        real(kind=rk) :: pdfTmp( layout%fStencil%QQ ) ! temporary local pdf values
+        real(kind=rk) :: rho, feqPlus, feqMinus, fPlus, fMinus, magicParam
+        real(kind=rk) :: d_omega, aux_omega, nu
+        real(kind=rk) :: transVel( nSolve*3 ) ! velocity from the transport field
+        real(kind=rk) :: uc, cqx2, cqy2, cqz2, cq2
+        real(kind=rk) :: a_e, a_xx, a_ww, a_xy, a_xz, a_yz
+        real(kind=rk) :: P_e, P_xx, P_ww, P_xy, P_xz, P_yz
+        integer :: vel_varPos ! position of transport velocity variable in varSys
+        real(kind=rk) :: inv_vel, u_fluid(3)
+        ! --------------------------------------------------------------------------
+        ! access scheme via 1st variable method data which is a state variable
+        call C_F_POINTER( varSys%method%val(derVarPos(1)%pdf)%method_Data, fPtr )
+        scheme => fPtr%solverData%scheme
+    
+        ! passive scalar has only one transport Variable
+        vel_varPos = scheme%transVar%method(1)%data_varPos
+        ! Get velocity field
+        call varSys%method%val(vel_varPos)%get_valOfIndex( &
+          & varSys  = varSys,                              &
+          & time    = params%general%simControl%now,       &
+          & iLevel  = level,                               &
+          & idx     = scheme%transVar%method(1)            &
+          &           %pntIndex%indexLvl(level)            &
+          &           %val(1:nSolve),                      &
+          & nVals   = nSolve,                              &
+          & res     = transVel                             )
+    
+        ! convert physical velocity into LB velocity
+        inv_vel = 1.0_rk / params%physics%fac( level )%vel
+        transVel = transVel * inv_vel
+
+        d_omega = fieldProp(1)%species%omega
+        magicParam = fieldProp(1)%species%lambda
```

Comment:
Why do you rename lambda to `magicParam` here? That's a source for confusion, just stick to lambda? Or make it analogous to omega and name it `d_lambda`.